### PR TITLE
ci: use octokit to add assignee

### LINKED ISSUE #6242: Port `take` workflow to use `oktokit`
Hi, I recommend using [GitHub scripts](https://github.com/actions/github-script) instead of `curl` for easier maintainance.

We can use simple js code here instead:

```js

octokit.rest.issues.addAssignees({
  owner,
  repo,
  issue_number,
  assignees,
});
```

ref: https://octokit.github.io/rest.js/v20#issues

_Originally posted by @Xuanwo in https://github.com/apache/arrow-rs/pull/6059#discussion_r1677710508_

### PR DESCRIPTION
**Pull request by dsgibbons**
Closes #6242.

Tested using:
- Repo: https://github.com/dsgibbons/test-action
- Issue: https://github.com/dsgibbons/test-action/issues/4
- Workflow run: https://github.com/dsgibbons/test-action/actions/runs/10438790900/job/28906516130

### CODE REVIEW

**Review by alamb — APPROVED** [review]
I looked at the examples and tests in other repositories. This looks good to me. 

cc @Xuanwo

**Comment by alamb** [comment]
https://github.com/apache/arrow-rs/actions/runs/10438818038/job/28962913511?pr=6267 failure is due to a infrastructure issue. I will retrigger it once the other CI has completed

File: .github/workflows/take.yml [hunk 1]
```diff
@@ -25,17 +25,16 @@ permissions:
...
-          else
-            echo "Cannot assign issue ${{ github.event.issue.number }} to ${{ github.event.comment.user.login }}"
-          fi
\ No newline at end of file
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - run: npm install @octokit/action
+      - run: node .github/actions/assign.mjs
```
Comment (Xuanwo) [inline .github/workflows/take.yml:36]: Compared to having our own implemented `assign.mjs`, I prefer using safer options like https://github.com/actions/github-script. It's easier to maintain and review.

What do you think?

Comment (Xuanwo) [reply .github/workflows/take.yml:36]: Makes sense to me. This PR itself is good enough to merge. I can help improve it later.

**Comment by alamb** [comment]
Thanks @dsgibbons  and @Xuanwo !

**Pull request merged**